File: 21558 - Nithin/week 3/1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        queue<TreeNode*> tree_queue;
        tree_queue.push(root);
        int level_sum, level = 1, qsize;
        int smallest_level, max_levelsum = INT_MIN;
        
        while(tree_queue.empty() == false){
            level_sum = 0;
            qsize = tree_queue.size();
            while(qsize != 0){
                TreeNode *node = tree_queue.front();
                tree_queue.pop();
                level_sum += node->val;
                if (node->left != NULL)
                    tree_queue.push(node->left);
                if (node->right != NULL)
                    tree_queue.push(node->right);
                --qsize;
            }
            if(level_sum > max_levelsum){
                smallest_level = level;
                max_levelsum = level_sum;
            }
            ++level;
        }
        return smallest_level;
        
    }
};

File: 21553 - Sai Nandan D/Week 3/1161_Maximum Level Sum of a Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        max_total, nodes = root.val, [root]                
        max_level, level = 1, 1                
        
        while nodes:            
            total = 0
            level_nodes = []
            for node in nodes:                
                total += node.val
                if node.left: level_nodes.append(node.left)
                if node.right: level_nodes.append(node.right)                    
            if total > max_total:
                max_total = total
                max_level = level
            nodes = level_nodes
            level += 1
            
        return max_level
